src/main/java/com/exasol/sql/expression/BinaryArithmeticExpression.java

```diff
@@ -0,0 +1,72 @@
+package com.exasol.sql.expression;
+
+import com.exasol.util.AbstractTreeNode;
+
+/**
+ * This class represents a binary arithmetic expression.
+ * <p>
+ * Examples: +, -, *, /
+ * </p>
+ */
+public class BinaryArithmeticExpression extends AbstractTreeNode implements ValueExpression {
+    private final BinaryArithmeticOperator arithmeticOperator;
+    private final ValueExpression left;
+    private final ValueExpression right;
+
+    private BinaryArithmeticExpression(final BinaryArithmeticOperator arithmeticOperator, final ValueExpression left,
+            final ValueExpression right) {
+        this.arithmeticOperator = arithmeticOperator;
+        this.left = left;
+        this.right = right;
+        addChild(left);
+        addChild(right);
+    }
+
+    /**
+     * Create a new {@link BinaryArithmeticExpression} instance.
+     * 
+     * @param operator arithmetic operator represented by {@link BinaryArithmeticOperator}
+     * @param left left part of the expression
+     * @param right right part of the expression
+     * @return new {@link BinaryArithmeticExpression} instance
+     */
+    public static BinaryArithmeticExpression of(final BinaryArithmeticOperator operator, final ValueExpression left,
+            final ValueExpression right) {
+        return new BinaryArithmeticExpression(operator, left, right);
+    }
+
+    /**
+     * Get a string representation of a member of this enum class. For example, + represents an ADD operator.
+     *
+     * @return string representation of an arithmetic operator
+     */
+    public String getStringOperatorRepresentation() {
+        return this.arithmeticOperator.getStringOperatorRepresentation();
+    }
+
+    @Override
+    public void accept(final ValueExpressionVisitor visitor) {
+        visitor.visit(this);
+        this.left.accept(visitor);
+        visitor.addOperator(this);
```

Comment:
This is not okay, since it leaks part of the functionality of the visitor into this class. Check the `Comparison` class for a similar example, that avoids this.

Reply:
fixed